### PR TITLE
style: improve contact editor design

### PR DESCRIPTION
**Pull request by SuhasDissa**
| Before | After |
| --- | --- |
| Screenshot_20240103_132844 | Screenshot_20240103_164439 |

### CODE REVIEW

**Review by M00NJ — APPROVED** [review]

File: app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import com.bnyro.contacts.obj.TranslatedType
+import com.bnyro.contacts.obj.ValueWithType
+
+@Composable
+fun SnapshotStateList<MutableState<ValueWithType>>.TextFieldGroup(
```
Comment (Bnyro) [inline app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt:30]:
```suggestion
fun TextFieldGroup(
```

Comment (Bnyro) [reply app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt:30]: I would prefer passing this as a normal argument, I don't think it's common to use extension functions for composables like that.

File: app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldEditor.kt [hunk 2]
```diff
@@ -21,28 +39,72 @@ fun TextFieldEditor(
...
+        state = textState,
+        keyboardType = keyboardType,
+        imeAction = imeAction,
+        leadingIcon = leadingIcon,
+        trailingIcon = if (showDeleteAction) ({
+            IconButton(onClick = { onDelete.invoke() }) {
+                Icon(
+                    imageVector = Icons.Rounded.Remove, contentDescription = stringResource(
+                        id = R.string.delete
+                    ), tint = Color.Red
```
Comment (Bnyro) [inline app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldEditor.kt:62]:
```suggestion
                    ), tint = MaterialTheme.colorScheme.error
```

**Comment by Bnyro** [comment]
You probably forgot to re-add the account type dialog?

**Comment by Bnyro** [comment]
I really like the visual changes btw.

My only concern is thr color of the text fields, they loom a bit to gray-ish to me personally.

**Comment by SuhasDissa** [comment]
> You probably forgot to re-add the account type dialog?

I removed the dialog and added a drop-down menu inside the top app bar. It's not visible if there's only one account present.
IMO, It's better to have the account switch in a more prominent place

**Comment by SuhasDissa** [comment]
> My only concern is thr color of the text fields, they loom a bit to gray-ish to me personally.

It's the default color of material 3 text field. Changing that is quite frustrating

File: app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                            Icon(
+                                imageVector = leadingIcon,
+                                contentDescription = null
+                            )
+                        }
+                    },
+                    shape = if (index == 0) RoundedCornerShape(50, 50, 0, 0) else RectangleShape
+                )
+            }
+            AnimatedVisibility(true) {
```
Comment (Bnyro) [inline app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt:72]: What's that line used for?

Comment (SuhasDissa) [reply app/src/main/java/com/bnyro/contacts/ui/components/editor/TextFieldGroup.kt:72]: Some mistake :)

**Pull request merged**